### PR TITLE
Deploy assets using Superchain ERC20

### PR DESCRIPTION
**Pull request by bradleycamacho**
Closes https://github.com/ethereum-optimism/devrel/issues/381

This doc explains how to deploy assests using the `SuperchainERC20` standard.

The issue had minimal collateral, so there may be pieces missing; especially regarding code examples and how we want to message this as a company. I've left various comments where I think specific review is needed.

### CODE REVIEW

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+### Deploy the `SuperchainERC20` Token Contract
+
+To ensure fungibility across chains, the SuperchainERC20 assets need to have the same contract address on all chains. Achieving this requires deterministic deployment methods, such as:
+
+* `Create2Deployer`: A smart contract that enables deploying contracts with predictable addresses.
+* `OptimismSuperchainERC20Factory`: A factory contract designed for L1-native tokens to ensure uniform deployments across chains.
+
+There are [many ways to do this](https://github.com/Arachnid/deterministic-deployment-proxy), but here's an example smart contract to start:
+
+```solidity
```
Comment (bradleycamacho) [inline pages/stack/interop/deploy-superchain-erc20.mdx:33]: We need an example here, but I wasn't able to find a good implementation of it. I use AI to create this, but we should absolutely replace this with something better.

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+# Issuing new assets with SuperchainERC20
+
+<Callout>
+  Interop is currently in active development and not yet ready for production use. The information provided here may change. Check back regularly for the most up-to-date information.
+</Callout>
+
+This guide explains how to issue new assets with the `SuperchainERC20` standard and bridge them effectively using the `SuperchainERC20Bridge`. If you want more information about the `SuperchainERC20 standard`, see our [`SuperchainERC20` standard explainer](/stack/interop/superchain-erc20)
+
+An important thing to note, is that bridging assets through the Superchain using SuperchainERC20 never affects the supply of your asset. Supply is always fixed; bridging and moving your asset across the superchain only affects which chain its located on.
```
Comment (bradleycamacho) [inline pages/stack/interop/deploy-superchain-erc20.mdx:18]: I added this based on the #docs message earlier today. Please let me know if we should move or rephrase. (Or remove!)

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+address tokenAddress = Create2Deployer.deploy(salt, bytecode, constructorArgs);
+```
+
+By deploying assets at identical addresses across multiple chains, we abstract away the complexity of cross-chain validation.
+
+### Implement the ICrosschainERC20 interface
+
+Ensure that your token contract conforms to the `ICrosschainERC20` interface to support cross-chain minting and burning. 
+
+```solidity
```
Comment (bradleycamacho) [inline pages/stack/interop/deploy-superchain-erc20.mdx:45]: I stole the comments from the specs page. We might want to expand on why the events "MUST" trigger

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  uint256 _source = L2ToL2CrossChainMessenger.crossDomainMessageSource();
+
+  _token.crosschainMint(_to, _amount);
+
+  emit RelayedERC20(address(_token), _from, _to, _amount, _source);
+}
+```
+</Steps>
+
+## Next steps
```
Comment (bradleycamacho) [inline pages/stack/interop/deploy-superchain-erc20.mdx:90]: Are these the correct places to send folks to?

File: pages/stack/interop/deploy-superchain-erc20.mdx (hunk 2 above)
Comment (zainbacchus) [reply pages/stack/interop/deploy-superchain-erc20.mdx:18]: @bradleycamacho  wdyt about 

An important thing to note is using `crosschainBurn` and `crosschainMint` on the ` SuperchainERC20`to move your asset across the Superchain only affects which chain it is located on and does not change the overall supply of the token. This keeps the token’s total amount the same across all networks, ensuring its value stays stable during the move and that the `SuperchainERC20` retains a unified, global supply count.

Comment (bradleycamacho) [reply pages/stack/interop/deploy-superchain-erc20.mdx]: Fixed in latest commit

**Comment by bradleycamacho** [comment]
@zainbacchus this is ready for your review whenever you have a second!

File: pages/stack/interop/deploy-superchain-erc20.mdx (hunk 1 above)
Comment (zainbacchus) [reply pages/stack/interop/deploy-superchain-erc20.mdx:33]: Here's ane xample:  https://github.com/ethereum-optimism/superchainerc20-starter/blob/main/packages/contracts/src/L2NativeSuperchainERC20.sol

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+```solidity
+// Example token deployment using Create2Deployer
+bytes32 salt = keccak256(abi.encodePacked("SuperchainToken"));
+address tokenAddress = Create2Deployer.deploy(salt, bytecode, constructorArgs);
+```
+
+By deploying assets at identical addresses across multiple chains, we abstract away the complexity of cross-chain validation.
+
+### Implement the ICrosschainERC20 interface
```
Comment (zainbacchus) [inline pages/stack/interop/deploy-superchain-erc20.mdx:41]: [IERC7802](https://specs.optimism.io/interop/token-bridging.html#ierc7802)

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+Ensure that your token contract conforms to the `ICrosschainERC20` interface to support cross-chain minting and burning.
```
Comment (zainbacchus) [inline pages/stack/interop/deploy-superchain-erc20.mdx:43]: [IERC7802](https://specs.optimism.io/interop/token-bridging.html#ierc7802)

File: pages/stack/interop/deploy-superchain-erc20.mdx (hunk 3 above)
Comment (zainbacchus) [reply pages/stack/interop/deploy-superchain-erc20.mdx:45]: I would just point here https://specs.optimism.io/interop/token-bridging.html#implementation 

so we can ensure everythign stays up to date

File: pages/stack/interop/deploy-superchain-erc20.mdx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import { Callout } from 'nextra/components'
+import { Steps } from 'nextra/components'
+
+# Issuing new assets with SuperchainERC20
+
+<Callout>
+  Interop is currently in active development and not yet ready for production use. The information provided here may change. Check back regularly for the most up-to-date information.
+</Callout>
+
+This guide explains how to issue new assets with the `SuperchainERC20` standard and bridge them effectively using the `SuperchainERC20Bridge`. If you want more information about the `SuperchainERC20 standard`, see our [`SuperchainERC20` standard explainer](/stack/interop/superchain-erc20)
```
Comment (zainbacchus) [inline pages/stack/interop/deploy-superchain-erc20.mdx:16]: remove word "standard"

**Comment by zainbacchus** [comment]
General feedback, for any code pls point the user to the relevant specs section https://specs.optimism.io/interop/token-bridging.html#implementation

**Review by zainbacchus — CHANGES_REQUESTED** [review]

**Review by zainbacchus — APPROVED** [review]
lgtm

**Review by cpengilly — APPROVED** [review]

**Pull request merged**